x-pack/plugins/actions/server/get_create_task_runner_function.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ActionType, Services } from './types';
+import { TaskInstance } from '../../task_manager';
+import { EncryptedSavedObjectsPlugin } from '../../encrypted_saved_objects';
+import { validateActionTypeConfig } from './validate_action_type_config';
+import { validateActionTypeParams } from './validate_action_type_params';
+
+interface CreateTaskRunnerFunctionOptions {
+  getServices: (basePath: string) => Services;
+  actionType: ActionType;
+  encryptedSavedObjectsPlugin: EncryptedSavedObjectsPlugin;
+}
+
+interface TaskRunnerOptions {
+  taskInstance: TaskInstance;
+}
+
+export function getCreateTaskRunnerFunction({
+  getServices,
+  actionType,
+  encryptedSavedObjectsPlugin,
+}: CreateTaskRunnerFunctionOptions) {
+  return ({ taskInstance }: TaskRunnerOptions) => {
+    return {
+      run: async () => {
+        const { namespace, id, actionTypeParams } = taskInstance.params;
+        const action = await encryptedSavedObjectsPlugin.getDecryptedAsInternalUser('action', id, {
+          namespace,
+        });
+        const mergedActionTypeConfig = {
+          ...(action.attributes.actionTypeConfig || {}),
+          ...(action.attributes.actionTypeConfigSecrets || {}),
+        };
+        const validatedActionTypeConfig = validateActionTypeConfig(
```

Comment:
__concern:__ hmm, I'm a bit concerned that Joi validation message may include the secret value if it fails validation.... @mikecote can you please show the error message we throw (and hence most likely return with 400 Bad Request or log somewhere) in this case? For example if we have a following schema for one of the secret attributes and the value doesn't start with a slash:
```ts
secret: Joi.string().regex(/^\//, `start with a slash`)
```

Reply:
Great catch, it the error would be `actionTypeConfig invalid: child "secret" fails because ["secret" with value "foo" fails to match the start with a slash pattern]` in this scenario. Should we just throw a generic error instead?

Reply:
Potentially we could analyze Joi error and see which fields failed and tweak the message (`blah-blah {field names that failed validation} failed validation`), but it feels a bit too much to me to be honest, your call though. I'd rather throw a generic error instead, it may affect dev experience for sure, but I'd feel safer. @kobelb do you think I'm exaggerating here? :slightly_smiling_face:

Reply:
>  @kobelb do you think I'm exaggerating here? 🙂

I don't think you're exaggerating, and this is a great catch. Joi's validation errors have disclosed the values of objects in the past, so throwing a generic error here (as long as it provides enough detail to fix the issue) is a great idea.

Reply:
Thanks, I pushed a fix for this :)